Fmc connections and constraints

AndrDragomir opened the pull request:
- Added fmc connection files for all carriers 
- Added an example file for eval board fmc connections
- Added the script that generates constraints based on the two types of connection files

AndreiGrozav commented:
In the file for eval board fmc connections, it would be nice tho have an extra column for pull up pull down since it is pretty common to use this feature.

AndreiGrozav commented:
This is up for debate.
Should the fmc_constr_generator.tcl be renamed in adi_fmc_constr_generator.tcl? To keep a consistency in scripts naming.
Keeping in mind that this script is sourced(relative path) not executed, and having another script in the OS environment path with the same name, will not affect things.

AndreiGrozav commented on projects/scripts/fmc_constr_generator.tcl:
```diff
@@ -0,0 +1,276 @@
+proc gen_fmc_constr {{fmc_index1 fmc} {fmc_index2 {}}} {
+  
+  set project_name [get_property NAME [current_project]]
```
Typically when we add support for a new project we do not start from an opened Vivado project.
If you see a value in using this script in an opened project, I suggest adding it as an extra feature not a base or a starting point.

The name of the carrier and daughter boards can be extracted from the current working directory.
set cwd [pwd]
set carrier [file tail $cwd]
set eval_board [file tail [file dirname $cwd]]

Also, in the current format of the code the script will fail.
e.g.: no files matched glob pattern "../../common/evb_zc706/evb_zc706_fmc_hpc*.txt"

AndreiGrozav commented on projects/scripts/fmc_constr_generator.tcl:
```diff
@@ -0,0 +1,276 @@
+  set constr_file [open "fmc_constr.xdc" w+]
+  
+  if {[string length $fmc_index2] == 0} {
+    set eval_board_path [glob ../common/$eval_board\_fmc*.txt]
```
Keep in mind that the project name is not always the same with the _bd.tcl
I would go with set eval_board_path [glob ../common/*_fmc*.txt]

AndreiGrozav commented on projects/scripts/fmc_constr_generator.tcl:
```diff
@@ -0,0 +1,276 @@
+      }
+    }
+  close $constr_file
+  add_files -fileset constrs_1 -norecurse fmc_constr.xdc
```
This is also a command that considers we have an opened vivado project. Having it as an extra feature is the way to go, in my opinion.

AndrDragomir commented:
v2: updated the core algorithm
      updated the common format for eval board files
      solved some of the discussed issues
      still needs some improvements

AndrDragomir commented:
v3: The script is now compatible with Intel platforms

AndrDragomir commented:
v4: Decided on a common format for terminations:
     - Always DIFF_TERM TRUE instead of DIFF_TERM_ADV TERM_100

IuliaCMoldovan reviewed (approved):
Seems good to me!

AndreiGrozav commented:
To keep consistency rename fmc_constr_generator.tcl to adi_fmc_constr_generator.tcl.
Add the documentation from your commit of the tcl file and some usage examples as the ones I added when running/sourcing the tcl file in https://wiki.analog.com/resources/fpga/docs/hdl/porting_project_quick_start_guide
Create a new section, like "Tips" with the above and also add a link to the .xls file in hdl/docs you created.

AndrDragomir commented:
v5: renamed the script to adi_fmc_constr_generator.tcl
Added the documentation to https://wiki.analog.com/resources/fpga/docs/hdl/porting_project_quick_start_guide
